### PR TITLE
[JENKINS-51082] - Avoid attaching calling stacktrace as that is done by Channel already

### PR DESCRIPTION
**Pull request by olivergondza**
See [JENKINS-51082](https://issues.jenkins-ci.org/browse/JENKINS-51082).

### Proposed changelog entries

* Do not duplicate caller stacktrace when FilePath#act fails

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests

### Desired reviewers

@oleg-nenashev as a reviewer of the original remoting enhancement.

### CODE REVIEW

File: core/src/main/java/hudson/FilePath.java [hunk 1]
```diff
@@ -1021,11 +1021,6 @@ public void copyFrom(FileItem file) throws IOException, InterruptedException {
                 return channel.call(wrapper);
             } catch (TunneledInterruptedException e) {
                 throw (InterruptedException)new InterruptedException(e.getMessage()).initCause(e);
-            } catch (AbortException e) {
-                throw e;    // pass through so that the caller can catch it as AbortException
```
Comment (olivergondza) [inline core/src/main/java/hudson/FilePath.java]: This one is safe to remove after the `IOException` handling `catch` is gone.

**Review by dwnusbaum — COMMENTED** [review]
Looks fine to me, although I think it would be good to add a test that with a custom FileCallable that throws an exception so we can test that the result matches what is desired.

**Review by jeffret-b — CHANGES_REQUESTED** [review]
I second the request for a test, if that can be reasonably added.

**Review by oleg-nenashev — APPROVED** [review]
Looks good. I do not consider missing tests as a blocker here, but let's keep it on hold for now

**Comment by olivergondza** [comment]
I am not quite sure how to capture the attributes of readability in the test. Here is the visual difference of the stacktrace:

https://gist.github.com/olivergondza/85f870c8a6b8f81b7198e9377df50a56/revisions#diff-39859e34161f0cb6f5f50974282a2593

**Comment by dwnusbaum** [comment]
@olivergondza I meant have a test that uses `Throwable#getCause` and `Throwable#getSuppressed` to check that the structure of the `Exception` is as expected, i.e. the following cause does not exist:

    Caused by: java.io.IOException: remote file operation failed: /home/ogondza/code/jenkins/jenkins/core/target/junit4020951958098692638/junit9094057646393124581.tmp at hudson.remoting.Channel@1e65ec8d:This side of the channel: java.io.IOException: Remote outer	
      at hudson.FilePath.act(FilePath.java:1028)	
      at hudson.FilePath.act(FilePath.java:1010)	
      at hudson.FilePathTest.simulateFailure(FilePathTest.java:863)	
      ... 26 more

I don't feel strongly enough about adding a test to block the PR, but if we care about the exception format it seems like a good idea to add a regression test.

**Comment by olivergondza** [comment]
That part is already covered in remoting, where the improvement is coming from: https://github.com/jenkinsci/remoting/pull/189/files#diff-9f5a92e09f1a02bae1e61ce82917f317. This PR merely removes now redundant behavior from core.

**Comment by olivergondza** [comment]
Do we still believe we need a test?

**Comment by jeffret-b** [comment]
I believe tests are valuable. I believe they help demonstrate the intended behavior. In writing the test I proposed I was more clearly able to see the proposed change. I believe tests are useful in helping to prevent regressions.

I think a test could be useful in this case. I can kind of see the argument in not having one here, but I think more is gained by having one.

But, I'm not going to insist on one here.

**Review by jeffret-b — APPROVED** [review]

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**